views/application.py

```diff
@@ -106,28 +104,42 @@ class ApplicationHiddenDetailView(generics.GenericAPIView):
 
     permission_classes = [permissions.IsAuthenticated, ]
 
-    def post(self,request,*args,**kwargs):
+    def post(self, request, *args, **kwargs):
 
         """
         Authenticate if the requesting user is Team Member
         of the application and check the password entered.
         :param request: the request being processed
         :param args: arguments
         :param kwargs: keyword arguments
-        :return:response containing client_secret
+        :return: response containing client_secret
         """
 
         try:
             application = Application.objects.get(pk = request.data.get('id'))
-            application_team_members = application.team_members.all()
-            for team_member in application_team_members:
-                if team_member.id == request.person.id:
-                    if request.user.check_password(request.data.get('password')):
-                        return response.Response(ApplicationHiddenDetailSerializer(application).data)
-                    else:
-                        return response.Response("Wrong Password",status=403)
+
+            if application.team_members.filter(id = request.person.id).exists():
+
+                if request.user.check_password(request.data.get('password')):
```

Comment:
@pradumangoyal the `check_password` import is used here.

Reply:
This is not an import, check_password is a property of the user model.

Reply:
Yes right.
I got that wrong.